### ISSUE TITLE
Rename get*() and get*List() methods

### ISSUE THREAD

**Issue by phenaproxima** [grom358/pharborist#22]
Not sure if this is possible, but purely from a usability standpoint the difference between the two is confusing. Let's take FunctionTrait::getParameters() and FunctionTrait::getParameterList() as examples -- the difference between the two is vague. What if these, and similar methods, were refactored into something like this:

$function->getParameters()->asArray();

getParameters() would return the ParameterListNode, and asArray() would do what ParameterListNode::getParameters() currently does.

So, in other words, rename FunctionTrait::getParameterList() to getParameters(), and ParameterListNode::getParameters() to asArray(). Similar changes could then be made for ArgumentListNode, FunctionCallNode, ClassMethodCallNode, and ObjectMethodCallNode.
